### PR TITLE
2 get topics

### PR DESCRIPTION
**Pull request by matt12171**
-Added further instructions to readme
-Added basic error handling
-Added get-topics with testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    return db.end()
+})
+
+describe('/api/topics', ()=> {
+    test('200: responds with 200 status code and an array of topic objects', ()=> {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                body.topics.forEach((topic) => {
```
Comment (2202Hannah) [inline __tests__/app.test.js:21]: Currently, this test would pass if it received an empty array. How can we make sure that the array is populated with the data we expect and not an empty array?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const { getTopics } = require('./controllers/topic.controller')
+const {
+    handlepsqlErrors,
+    handleCustomErrors,
+    handleServerErrors,
+  } = require("./errors");
+
+const app = express()
+
+app.use(express.json())
```
Comment (2202Hannah) [inline app.js]: We don't technically need this middleware function yet as we only have a get request. Please remove before you merge to main.

File: errors/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+exports.handlepsqlErrors = (err, req, res, next) => {
```
Comment (2202Hannah) [inline errors/index.js]: We don't currently have any PSQL or custom errors that are handled in our code - please remove these middleware functions until they are needed in your code.

File: models/topic.model.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const db = require('../db/connection')
+const format = require('pg-format')
```
Comment (2202Hannah) [inline models/topic.model.js]: You are not currently using pg-format in your topics model - please remove this until it is needed.

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hey :) really nice work on this ticket so far. Your test suite is set up well and you have made sure to seed the database before each test. You are also using the MVC structure which is great to see.

I have made some in-line comments suggesting to remove certain bits of code that are not currently needed (such as middleware functions and packages that are required in and not used). 

You will also need to make some changes to your test to make sure it doesn't pass unexpectedly if it receives an empty array. Once these changes have been made please make another pr for this ticket. Thanks :)

**Review by mezzDavies — APPROVED** [review]
Good stuff on removing the unneeded code.  There is no need to _resubmit_ a PR though unless specifically asked to do so. Please merge and move onto the next task 👍
